### PR TITLE
chore: update all licenses to latest

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

Updated all the license information in the repository to use Elastic License.

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-2057/update-all-license-files-in-the-repo-to-elastic-license-js-sdk

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **Documentation**
	- Updated `README.md` files across various packages to include sections for License, Contribute, Contact Us, Follow Us, and Our Supporters.
	- Added licensing information under Elastic License 2.0 in `LICENSE.md` files across multiple packages.
	- Deprecated the RudderStack JavaScript SDK package and updated references to the latest package in `packages/analytics-v1.1/README.md`.

- **New Features**
	- Introduced clearer licensing and contribution guidelines for better user engagement and transparency.

### CODE REVIEW

File: packages/analytics-js-service-worker/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance ##
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-service-worker/LICENSE.md]: Done.

File: packages/analytics-js-plugins/LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance ##
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/LICENSE.md]: Done.

File: packages/analytics-js-integrations/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance ##
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-integrations/LICENSE.md]: Done.

File: packages/analytics-js-cookies/LICENSE.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance ##
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-cookies/LICENSE.md]: Done.

File: packages/analytics-js-common/LICENSE.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance ##
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-common/LICENSE.md]: Done.

File: packages/analytics-js-plugins/README.md [hunk 6]
```diff
@@ -40,3 +40,47 @@ Plugins are JavaScript SDK v3 features that you can optionally load on demand. A
...
+
+[rudderstack-blog]: https://rudderstack.com/blog/
+[slack]: https://resources.rudderstack.com/join-rudderstack-slack
+[twitter]: https://twitter.com/rudderstack
+[linkedin]: https://www.linkedin.com/company/rudderlabs/
+[devto]: https://dev.to/rudderstack
+[medium]: https://rudderstack.medium.com/
+[youtube]: https://www.youtube.com/channel/UCgV-B77bV_-LOmKYHw8jvBw
+[hackernews]: https://news.ycombinator.com/item?id=21081756
+[producthunt]: https://www.producthunt.com/posts/rudderstack
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/README.md]: Done.

File: packages/analytics-js/LICENSE.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+*you* refers to the individual or entity agreeing to these terms.
+
+*your company* is any legal entity, sole proprietorship, or other kind of organization that you work for, plus all organizations that have control over, are under the control of, or are under common control with that organization. *control* means ownership of substantially all the assets of an entity, or the power to direct its management and policies by vote, contract, or otherwise. Control can be direct or indirect.
+
+*your licenses* are all the licenses granted to you for the software under these terms.
+
+*use* means anything you do with the software requiring one of your licenses.
+
+*trademark* means trademarks, service marks, and similar rights.
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/LICENSE.md]: Done.

File: packages/analytics-v1.1/README.md [hunk 8]
```diff
@@ -312,10 +314,46 @@ For examples and specific details look into:
...
+## License
+
+This project is licensed under the terms of the Elastic License 2.0. Please see the [LICENSE.md](LICENSE.md) file for license rights and limitations. We recommend that you review the license terms to understand the permissions and restrictions that apply.
+
+If you have any questions about licensing, please [contact us](#contact-us) or refer to the [official Elastic licensing](https://www.elastic.co/licensing/elastic-license) page.
+
 ## Contribute
 
-We would love to see you contribute to this project. Get more information on how to contribute [**here**](./CONTRIBUTING.md).
+We invite you to contribute to this project. For more information on how to contribute, please see [**here**](../../CONTRIBUTING.md).
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-v1.1/README.md:325]: It is correct.

File: LICENSE.md [hunk 9]
```diff
@@ -1,36 +1,36 @@
...
-## Termination ## 
+## Termination
 If you use the software in violation of these terms, such use is not licensed, and your licenses will automatically terminate. If the licensor provides you with a notice of your violation, and you cease all violation of this license no later than 30 days after you receive that notice, your licenses will be reinstated retroactively. However, if you violate these terms after such reinstatement, any additional violation of these terms will cause your licenses to terminate automatically and permanently.
 
-## No Liability ## 
+## No Liability
 As far as the law allows, the software comes as is, without any warranty or condition, and the licensor will not be liable to you for any damages arising out of these terms or the use or nature of the software, under any kind of legal claim.
 
-## Definitions ## 
+## Definitions
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply LICENSE.md]: Done.

File: packages/analytics-v1.1/LICENSE.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-v1.1/LICENSE.md:3]: Done.

File: packages/analytics-js/LICENSE.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/LICENSE.md:3]: Done.

File: packages/analytics-js-service-worker/LICENSE.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-service-worker/LICENSE.md:3]: Done.

File: packages/analytics-js-plugins/LICENSE.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
+### Elastic License 2.0 (ELv2)
+
+## Acceptance
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/LICENSE.md:3]: Done.

File: packages/analytics-v1.1/README.md [hunk 14]
```diff
@@ -36,7 +38,7 @@ The JavaScript SDK lets you track customer event data from your website and send
 - [**Usage in Chrome Extensions**](#usage-in-chrome-extensions)
 - [**Usage in Serverless Runtimes**](#usage-in-serverless-runtimes)
 
-| **IMPORTANT**: We have deprecated the service worker export from RudderStack JavaScript SDK npm package and decoupled it to a new package. <br/>If you still wish to use it for your project, refer to [**@rudderstack/analytics-js-service-worker package**](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker). |
+| **IMPORTANT**: We have deprecated the service worker export from RudderStack JavaScript SDK NPM package and decoupled it to a new package. <br/>If you still wish to use it for your project, refer to [**@rudderstack/analytics-js-service-worker package**](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker). |
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-v1.1/README.md]: Done.

File: packages/analytics-v1.1/README.md [hunk 15]
```diff
@@ -20,6 +20,8 @@
 ---
 
 # RudderStack JavaScript SDK
+| :warning: This package is deprecated. Please switch to the latest [@rudderstack/analytics-js](https://www.npmjs.com/package/@rudderstack/analytics-js) package. For more details visit, https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/. |
+| :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-v1.1/README.md]: Done.

File: packages/analytics-v1.1/README.md [hunk 16]
```diff
@@ -312,10 +314,46 @@ For examples and specific details look into:
 - [Vercel Edge Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/serverless/USAGE.md)
 - [Cloudflare Worker Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/serverless/USAGE.md)
 
+## License
+
+This project is licensed under the terms of the Elastic License 2.0. Please see the [LICENSE.md](LICENSE.md) file for license rights and limitations. We recommend that you review the license terms to understand the permissions and restrictions that apply.
+
+If you have any questions about licensing, please [contact us](#contact-us) or refer to the [official Elastic licensing](https://www.elastic.co/licensing/elastic-license) page.
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-v1.1/README.md]: Done.

File: packages/analytics-js-plugins/README.md [hunk 17]
```diff
@@ -40,3 +40,47 @@ Plugins are JavaScript SDK v3 features that you can optionally load on demand. A
...
+
+## License
+
+This project is licensed under the terms of the Elastic License 2.0. Please see the [LICENSE.md](LICENSE.md) file for license rights and limitations. We recommend that you review the license terms to understand the permissions and restrictions that apply.
+
+If you have any questions about licensing, please [contact us](#contact-us) or refer to the [official Elastic licensing](https://www.elastic.co/licensing/elastic-license) page.
+
+## Contribute
+
+We invite you to contribute to this project. For more information on how to contribute, please see [**here**](../../CONTRIBUTING.md).
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/README.md]: Done.

File: package.json [hunk 18]
```diff
@@ -51,7 +51,7 @@
     "deploy:npm": "nx affected --target=deploy --parallel=1 --skipCommitTypes=docs,ci,chore,test"
   },
   "author": "RudderStack",
-  "license": "MIT",
+  "license": "SEE LICENSE IN LICENSE.md",
```
Comment (arnab-p) [inline package.json:54]:
```suggestion
  "license": "ELv2",
```

Please make similar changes accordingly

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: package.json (hunk 18 above)
Comment (saikumarrs) [reply package.json]: Done.

Comment (gitcommitshow) [reply package.json]: The correct SPDX identifier is - `Elastic-2.0`
Reference - https://spdx.org/licenses/

Comment (saikumarrs) [reply package.json]: Addressed.

**Review by arnab-p — APPROVED** [review]

**Review by ameypv-rudder — APPROVED** [review]

**Review by gitcommitshow — APPROVED** [review]

**Pull request merged**